opaque_keys/edx/locator.py

```diff
@@ -193,8 +193,9 @@ def __init__(self, org=None, course=None, run=None, branch=None, version_guid=No
             if version_guid:
                 version_guid = self.as_object_id(version_guid)
 
-            if not all(field is None or self.ALLOWED_ID_RE.match(field) for field in [org, course, run, branch]):
-                raise InvalidKeyError(self.__class__, [org, course, run, branch])
+            for field in [org, course, run, branch]:
+                if not (field is None or self.ALLOWED_ID_RE.match(field)):
+                    raise InvalidKeyError(self.__class__, "Special characters not allowed in '{}'".format(field))
```

Comment:
Optional: It might be good to include the name of the field that the invalid string is passed in.